tests/net/socket/misc/src/main.c

```diff
@@ -421,13 +423,161 @@ void test_ipv6_getpeername(void)
 	test_getpeername(AF_INET6);
 }
 
+void test_getsockname_tcp(int family)
+{
+	int ret;
+	int sock_c;
+	int sock_s;
+	struct sockaddr local_addr = { 0 };
+	socklen_t local_addr_len = ADDR_SIZE(family);
+	struct sockaddr srv_addr = { 0 };
+
+	srv_addr.sa_family = family;
+	if (family == AF_INET) {
+		net_sin(&srv_addr)->sin_port = htons(DST_PORT);
+		ret = inet_pton(AF_INET, TEST_MY_IPV4_ADDR,
+				&net_sin(&srv_addr)->sin_addr);
+	} else {
+		net_sin6(&srv_addr)->sin6_port = htons(DST_PORT);
+		ret = inet_pton(AF_INET6, TEST_MY_IPV6_ADDR,
+				&net_sin6(&srv_addr)->sin6_addr);
+	}
+	zassert_equal(ret, 1, "inet_pton failed");
+
+	sock_c = socket(family, SOCK_STREAM, IPPROTO_TCP);
+	zassert_true(sock_c >= 0, "socket open failed");
+	sock_s = socket(family, SOCK_STREAM, IPPROTO_TCP);
+	zassert_true(sock_s >= 0, "socket open failed");
+
+	/* Verify that unbound/unconnected socket has no local address set */
+	ret = getsockname(sock_c, &local_addr, &local_addr_len);
+	zassert_equal(ret, -1, "getsockname shouldn've failed");
+	ret = getsockname(sock_s, &local_addr, &local_addr_len);
+	zassert_equal(ret, -1, "getsockname shouldn've failed");
```

Comment:
Should we also make sure that errno is set correctly?

Reply:
I've added extra checks for errno.